go/libkb/loaduser.go

```diff
@@ -192,6 +192,8 @@ func LoadUser(arg LoadUserArg) (ret *User, err error) {
 		err = NoKeyError{emsg}
 	}
 
+	checkForNewTracksIfSelf(arg, ret.IDTable())
```

Comment:
the `LoadUser` call should short-circuit if it realizes that it has a fresh version (after the merkle leaf check). No reason to load the rest of the user just to throw it on the floor.

Similarly, use the freshness check in sigchain loading to deduce what the previously-save chain tail was, and what the current chain-tail is, and feel that data through to `checkForNewTrackIfSelf` to remove the need for redundant level-DB storage of this info.